### PR TITLE
Rename githash to build_version

### PR DESCRIPTION
**Pull request by tmrts**
Renames ambiguous `githash`.

### CODE REVIEW

File: docs/devel/releasing.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ oncall.
 Before proceeding to the next step:
 
 ```sh
-export GITHASH=v1.2.0-alpha.2.164+b44c7d79d6c9bb
+export gitcommithash=v1.2.0-alpha.2.164+b44c7d79d6c9bb
```
Comment (thockin) [inline docs/devel/releasing.md]: exported vars should be CAPS.  GIT_COMMIT_HASH ?

**Comment by thockin** [comment]
I don't have a problem being more precise, but is this really ambiguous?

File: docs/devel/releasing.md (hunk 1 above)
Comment (tmrts) [inline docs/devel/releasing.md]: You are right. It was lowercase in the previous versions upper case with underscore is better

**Comment by tmrts** [comment]
@thockin pedantically speaking, there are different types of hashes. But, I assumed it would be helpful since it was mentioned in a "TODO" item.

**Comment by thockin** [comment]
Like I said, I am fine with it.

File: docs/devel/releasing.md [hunk 2]
```diff
@@ -93,21 +93,21 @@ commits,) so choose the latest build on the release branch.  (Remember, that
 branch should be frozen.)
 
 Once you find some greens, you can find the git hash for a build by looking at
-the Full Console Output and searching for `githash=`. You should see a line:
+the Full Console Output and searching for `GIT_COMMIT_HASH=`. You should see a line:
```
Comment (thockin) [inline docs/devel/releasing.md]: Where does this string get printed?  I don't see a change in this PR that changes a printout from "githash" to "GIT_COMMIT_HASH"

**Comment by tmrts** [comment]
@thockin cool :+1:

File: docs/devel/releasing.md [hunk 3]
```diff
@@ -144,7 +144,7 @@ Decide what version you're cutting and export it:
 Then, run
 
 ```console
-./release/cut-official-release.sh "${VER}" "${GITHASH}"
+./release/cut-official-release.sh "${VER}" "${GIT_COMMIT_HASH}"
```
Comment (tmrts) [inline docs/devel/releasing.md]: @thockin this is the only place the exported variable is used as far as I can tell.

File: docs/devel/releasing.md (hunk 2 above)
Comment (thockin) [inline docs/devel/releasing.md]: Someone who has cut a release should probably explain how this works.  Is there a call to un-parameterized `set` or somethign that ends up dumping all the env vars?

@ihmccreery

File: docs/devel/releasing.md (hunk 3 above)
Comment (tmrts) [inline docs/devel/releasing.md]: @thockin  nevermind my previous comment. There are two different usages in the document, one should be "gitcommithash", the exported variable should be "GIT_COMMIT_HASH"

**Comment by thockin** [comment]
Assigning to Ike since it is his TODO

**Comment by ihmccreery** [comment]
Thanks for this @tmrts.

I'd much rather call it `build_version`/`BUILD_VERSION` than `gitcommithash`, because it's not actually anything related to `git` directly.  It's derived from [git describe](https://git-scm.com/docs/git-describe), but it's munged enough that it's worth calling it what it is.

One other comment, otherwise we're good.

File: docs/devel/releasing.md [hunk 4]
```diff
@@ -93,21 +93,21 @@ commits,) so choose the latest build on the release branch.  (Remember, that
 branch should be frozen.)
 
 Once you find some greens, you can find the git hash for a build by looking at
-the Full Console Output and searching for `githash=`. You should see a line:
+the Full Console Output and searching for `gitcommithash=`. You should see a line:
```
Comment (ihmccreery) [inline docs/devel/releasing.md]: Please add a note that for `v1.1` and below, it was called `githash`.

**Comment by tmrts** [comment]
@ihmccreery there is also another parameter used with githash called `version`. It might be confusing to call it just `build_version` since they both look very similar. How about `build_hash`?

**Comment by ihmccreery** [comment]
You're right, but we should change `version`, since that's totally nondescript.  How about `release-version` and `build-version`, with an explanation:

> release-version is the version you want to release, build-version is the version of the build you want to mark as `release-version`.  E.g., I might want to mark the build-version `v1.2.0-alpha.3.45+asdf` as the release version `v1.2.0-alpha.4`.

Make sense?  I think "___-hash" remains confusing, because `v1.2.0-alpha.3.45+asdf` isn't a hash, (`asdf` is the hash).

**Comment by tmrts** [comment]
I agree that is much better.

**Comment by ihmccreery** [comment]
Great, thanks!  Please squash, then I'll LGTM.

**Comment by tmrts** [comment]
@ihmccreery done thanks.

**Comment by tmrts** [comment]
Fixed `hack/verify-flags-underscore` error please re-apply.

**Pull request merged**